io.openems.edge.evcs.spelsberg.smart/src/io/openems/edge/evcs/spelsberg/smart/EvcsSpelsbergSmartImpl.java

```diff
@@ -0,0 +1,343 @@
+
+				new FC3ReadRegistersTask(1008, Priority.LOW,
+						m(EvcsSpelsbergSmart.ChannelId.CURRENT_L1, new UnsignedWordElement(1008))),
+
+				new FC3ReadRegistersTask(1010, Priority.LOW,
+						m(EvcsSpelsbergSmart.ChannelId.CURRENT_L2, new UnsignedWordElement(1010))),
+
+				new FC3ReadRegistersTask(1012, Priority.LOW,
+						m(EvcsSpelsbergSmart.ChannelId.CURRENT_L3, new UnsignedWordElement(1012))),
+
+				new FC3ReadRegistersTask(1020, Priority.HIGH,
+						m(Evcs.ChannelId.CHARGE_POWER, new UnsignedDoublewordElement(1020))),
+
+				new FC3ReadRegistersTask(1020, Priority.HIGH,
+						m(EvcsSpelsbergSmart.ChannelId.POWER_TOTAL, new UnsignedDoublewordElement(1020))),
+
+				new FC3ReadRegistersTask(1024, Priority.HIGH,
+						m(EvcsSpelsbergSmart.ChannelId.POWER_L1, new UnsignedDoublewordElement(1024))),
+
+				new FC3ReadRegistersTask(1028, Priority.HIGH,
+						m(EvcsSpelsbergSmart.ChannelId.POWER_L2, new UnsignedDoublewordElement(1028))),
+
+				new FC3ReadRegistersTask(1032, Priority.HIGH,
+						m(EvcsSpelsbergSmart.ChannelId.POWER_L3, new UnsignedDoublewordElement(1032))),
+
+				new FC3ReadRegistersTask(1036, Priority.LOW,
+						m(Evcs.ChannelId.ACTIVE_CONSUMPTION_ENERGY, new UnsignedDoublewordElement(1036))),
+
+				new FC3ReadRegistersTask(1100, Priority.LOW,
+						m(EvcsSpelsbergSmart.ChannelId.MAX_HARDWARE_CURRENT, new UnsignedWordElement(1100))),
+
+				new FC3ReadRegistersTask(1102, Priority.LOW,
+						m(EvcsSpelsbergSmart.ChannelId.MIN_HARDWARE_CURRENT, new UnsignedWordElement(1102))),
+
+				new FC3ReadRegistersTask(1502, Priority.HIGH,
+						m(EvcsSpelsbergSmart.ChannelId.CHARGE_ENERGY_SESSION, new UnsignedWordElement(1502))),
+
+				new FC3ReadRegistersTask(1504, Priority.LOW,
+						m(EvcsSpelsbergSmart.ChannelId.CHARGE_START_TIME, new UnsignedDoublewordElement(1504))),
+
+				new FC3ReadRegistersTask(1508, Priority.LOW,
+						m(EvcsSpelsbergSmart.ChannelId.CHARGE_DURATION_SESSION, new UnsignedDoublewordElement(1508))),
+
+				new FC3ReadRegistersTask(1512, Priority.LOW,
+						m(EvcsSpelsbergSmart.ChannelId.CHARGE_STOP_TIME, new UnsignedDoublewordElement(1512))),
+
+				new FC3ReadRegistersTask(2000, Priority.LOW,
+						m(EvcsSpelsbergSmart.ChannelId.CHARGE_SAVE_CURRENT_LIMIT, new UnsignedWordElement(2000))),
+
+				new FC3ReadRegistersTask(6000, Priority.HIGH,
+						m(EvcsSpelsbergSmart.ChannelId.LIFE_BIT, new UnsignedWordElement(6000))),
+
+				new FC6WriteRegisterTask(6000, m(EvcsSpelsbergSmart.ChannelId.LIFE_BIT, new UnsignedWordElement(6000))),
+
+				new FC16WriteRegistersTask(5000,
+						m(EvcsSpelsbergSmart.ChannelId.APPLY_CHARGE_POWER_LIMIT, new UnsignedDoublewordElement(5000))),
+
+				new FC6WriteRegisterTask(5004,
+						m(EvcsSpelsbergSmart.ChannelId.APPLY_CHARGE_CURRENT_LIMIT, new UnsignedWordElement(5004))));
+
+		addStatusCallback();
+		addPhaseDetectionCallback();
+		addPowerConsumptionCallback();
+		return modbusProtocol;
+	}
+
+	@Override
+	public String debugLog() {
```

Comment:
Reduce debug log.
If you need additional channels in the log, you could add them in the DebugLog Component with "Additional Channels" or using the DetailedDebugLog Component

Reply:
done